query-engine/connectors/sql-query-connector/src/filter/visitor.rs

```diff
@@ -213,260 +305,283 @@ impl AliasedCondition for ScalarFilter {
 
                 convert_scalar_filter(
                     comparable,
-                    self.condition,
-                    state.reverse(),
-                    self.mode,
+                    filter.condition,
+                    self.reverse(),
+                    filter.mode,
                     &[],
-                    state.alias(),
+                    self.parent_alias(),
                     false,
                     ctx,
                 )
             }
-            _ => scalar_filter_aliased_cond(self, state.alias(), state.reverse(), ctx),
+            _ => scalar_filter_aliased_cond(filter, self.parent_alias(), self.reverse(), ctx),
         }
     }
-}
 
-fn scalar_filter_aliased_cond(
-    sf: ScalarFilter,
-    alias: Option<Alias>,
-    reverse: bool,
-    ctx: &Context<'_>,
-) -> ConditionTree<'static> {
-    match sf.projection {
-        ScalarProjection::Single(field) => {
-            let comparable: Expression = field.aliased_col(alias, ctx).into();
+    fn visit_relation_filter(
+        &mut self,
+        filter: RelationFilter,
+        ctx: &Context<'_>,
+    ) -> (ConditionTree<'static>, Option<Vec<AliasedJoin>>) {
+        let parent_alias = self.parent_alias().map(|a| a.to_string(None));
 
-            convert_scalar_filter(comparable, sf.condition, reverse, sf.mode, &[field], alias, false, ctx)
-        }
-        ScalarProjection::Compound(fields) => {
-            let columns: Vec<Column<'static>> = fields
-                .clone()
-                .into_iter()
-                .map(|field| field.aliased_col(alias, ctx))
-                .collect();
+        match &filter.condition {
+            RelationCondition::NoRelatedRecord if self.should_render_join() && !filter.field.is_list() => {
```

Comment:
`{ to_one: { ... } }` -> Now uses a `LEFT JOIN`.

Reply:
We could add some documentation here, and some note as to why an `INNER JOIN` cannot _always_ be used and how we could optimize it.